regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
 
-    to_plot$faster_route <<- readRDS(file.path(region$data_dir, "rf.Rds" ))
-    to_plot$faster_route@data <<- cbind(
-      to_plot$faster_route@data[!(names(to_plot$faster_route) %in% names(to_plot$straight_line))],
-      to_plot$straight_line@data)
-    to_plot$quieter_route <<- readRDS(file.path(region$data_dir, "rq.Rds"))
-    to_plot$quieter_route@data <<- cbind(
-      to_plot$quieter_route@data[!(names(to_plot$quieter_route) %in% names(to_plot$straight_line))],
-      to_plot$straight_line@data)
+    # Identify that region repopulation has happened
+    region$repopulate_region <<- T
 
-    # Add rqincr column to the quiet data
-    to_plot$quieter_route@data$rqincr <<- to_plot$quieter_route@data$length / to_plot$faster_route@data$length
+    # Identify purposes and geographies available in region
+    purposes_list <- c("commute", "school", "alltrips")
+    region$purposes_present <<- (dir.exists(file.path(data_regional_root, purposes_list, "msoa", region$current)) | dir.exists(file.path(data_regional_root, purposes_list, "lsoa", region$current)))
+    geographies_list <- c("msoa", "lsoa")
+    region$geographies_present <<- dir.exists(file.path(data_regional_root, input$purpose, geographies_list, region$current))
+    update_purposegeog(region$purposes_present, region$geographies_present)
 
-    region$all_trips <- dir.exists(file.path(data_dir_root, region$current , 'all-trips'))
-    # Identify the centre of the current region
-    to_plot$center_dim <<- rgeos::gCentroid(regions[regions$Region == region$current,], byid = TRUE)@coords
-    region$repopulate_region <- T
+    # Identify the centre of the current region, save in to_plot
+    to_plot$center_dim <<- rgeos::gCentroid(regions[regions$region_name == region$current, ], byid = TRUE)@coords
 
-  })
+    # Load data to to_plot (if data exists - this varies by purpose/geography)
+    if (file.exists(file.path(region$data_dir, "z.Rds"))) {
+      to_plot$zones <<- readRDS(file.path(region$data_dir, "z.Rds"))
+    } else {
+      to_plot$zones <<- NULL
+    }
 
-  region <- reactiveValues(current = NA, data_dir = NA, repopulate_region = F, all_trips = NA)
-  show_no_lines <- c("none", "lsoa_base_map")
+    if (file.exists(file.path(region$data_dir, "c.Rds"))) {
+      to_plot$centroids <<- readRDS(file.path(region$data_dir,  "c.Rds"))
+    } else {
+      to_plot$centroids <<- NULL
+    }
 
-  observe({
-    output$production_branch <- renderText({ifelse(production_branch, "true", "false")})
-    # If a region does not have an 'all-trips'directory, hide the trip panel
-    if (!region$all_trips){
-      # hide trip_panel
-      shinyjs::hide("trip_panel")
-    }else{
-      # show trip_panel
-      shinyjs::show("trip_panel")
+    if (file.exists(file.path(region$data_dir, "l.Rds"))) {
+      to_plot$straight_lines <<- readRDS(file.path(region$data_dir, "l.Rds"))
+    } else {
+      to_plot$straight_lines <<- NULL
```

Comment:
I think we could make this a bit DRYer

```R
load_data <- function(to_plot, to_plot_subset, filepath|)
  if (file.exists(filepath)) {
    to_plot[to_plot_subset] <<- readRDS(filepath)
  } else {
    to_plot[to_plot_subset] <<- NULL
  }
)
# then to use it:
load_data(to_plot, "straight_lines", file.path(region$data_dir, "l.Rds"))
```

Reply:
Sounds good. Seeing as this PR has already been merged, are you up for making these changes?

Reply:
Yes, might take me a while though.

Reply:
yes, agreed this is better